scheduler/workflow.go

```diff
@@ -315,8 +315,10 @@ func (s *schedulerWorkflow) Start(t *task) {
 	// Block until the job has been either run or skipped.
 	errors := t.manager.Work(j).Promise().Await()
 
-	if len(errors) != 0 {
-		t.RecordFailure(j.Errors())
+	if len(errors) != 0 && len(j.Errors()) > 0 {
```

Comment:
Can we create an issue on this? It seems with Collect Jobs, we check the return `errors` but then pass `j.Errors()` into `RecordFailures` but for process and publish jobs, we pass `errors` into RecordFailures.